subprojects/core/src/main/java/org/gradle/api/internal/initialization/DefaultClassLoaderScope.java

```diff
@@ -115,7 +115,7 @@ private void buildEffectiveLoaders() {
                 }
             } else { // creating before locking, have to create the most flexible setup
                 if (Boolean.getBoolean(STRICT_MODE_PROPERTY)) {
-                    throw new IllegalStateException("Attempt to define scope class loader before scope is locked");
+                    throw new IllegalStateException("Attempt to define scope class loader before scope is locked " + id);
```

Comment:
:o: That would produce messages like `Attempt do define scope class loader before scope is locked ClassLoaderScopeIdentifier{...}` with just one space between the message and the identifier, how about something like `... before scope is locked, scope identifier is " + id"`